.clang-format

```diff
@@ -1,101 +1,209 @@
 ---
-Language:        Cpp
+Language: Cpp
 AccessModifierOffset: -2
-AlignAfterOpenBracket: AlwaysBreak
-AlignConsecutiveAssignments: false
-AlignConsecutiveDeclarations: true
-AlignConsecutiveMacros: true
-AlignEscapedNewlinesLeft: true
-AlignOperands:   true
+AlignAfterOpenBracket: DontAlign
+AlignConsecutiveMacros: Consecutive
+AlignConsecutiveAssignments: None
+AlignConsecutiveBitFields: None
+AlignConsecutiveDeclarations: None
+AlignEscapedNewlines: Left
+AlignOperands: DontAlign
 AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
 AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: false
+AllowShortEnumsOnASingleLine: true
+AllowShortBlocksOnASingleLine: Never
 AllowShortCaseLabelsOnASingleLine: false
-AllowShortFunctionsOnASingleLine: Inline
-AllowShortIfStatementsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortLambdasOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: Never
 AllowShortLoopsOnASingleLine: false
-AlwaysBreakAfterDefinitionReturnType: TopLevel
-AlwaysBreakAfterReturnType: All
+AlwaysBreakAfterReturnType: TopLevel
 AlwaysBreakBeforeMultilineStrings: false
-AlwaysBreakTemplateDeclarations: true
-BinPackArguments: false
+AlwaysBreakTemplateDeclarations: MultiLine
+
+# Only works wtih macros that do not take parameters
+AttributeMacros:
+ - HSE_ALWAYS_INLINE
+ - HSE_COLD
+ - HSE_CONST
+ - HSE_HOT
+ - HSE_MAYBE_UNUSED
+ - HSE_NOINLINE
+ - HSE_NORETURN
+ - HSE_PACKED
+ - HSE_READ_MOSTLY
+ - HSE_RETURNS_NONNULL
+ - HSE_SENTINEL
+ - HSE_USED
+ - HSE_WARN_UNUSED_RESULT
+ - HSE_WEAK
+
+# Treating function attribute macros that take parameters as type macros
+# helps format function declarations that use them.
+TypenameMacros:
+ - HSE_NONNULL
+ - HSE_PRINTF
+ - HSE_ALIGNED
+StatementAttributeLikeMacros:
```

Comment:
Is this not where you can put HSE_ALIGNED?

Reply:
Updated comments to explain it better.
```
# AttributeMacros only work wtih macros that do not take parameters.
# HSE attribute macros that take parameters work better as TypenameMacros
# because clang-format assumes they have parameters.
AttributeMacros:
 - HSE_ALWAYS_INLINE
 - HSE_COLD
 - HSE_CONST
 - HSE_HOT
 - HSE_MAYBE_UNUSED
 - HSE_NOINLINE
 - HSE_NORETURN
 - HSE_PACKED
 - HSE_READ_MOSTLY
 - HSE_RETURNS_NONNULL
 - HSE_SENTINEL
 - HSE_USED
 - HSE_WARN_UNUSED_RESULT
 - HSE_WEAK

# Treating function attribute macros that take parameters as TypenameMacros
# helps format function declarations that use them.
TypenameMacros:
 - HSE_NONNULL
 - HSE_PRINTF
 - HSE_ALIGNED
```